### PR TITLE
Fix LighthouseHostFactory argument are being ignored

### LINKED ISSUE #242: LighthouseHostFactory method arguments are being ignored
Right now, LighthouseHostFactory method arguments are being ignored/not used in code.

https://github.com/petabridge/lighthouse/blob/dev/src/Lighthouse/LighthouseHostFactory.cs#L28-L33

### PR DESCRIPTION
**Pull request by Arkatufus**
Fix #242

### Changes

Method argument are now being injected as the top most config using HOCON fallback.

### CODE REVIEW

File: src/Lighthouse/LighthouseHostFactory.cs [hunk 1]
```diff
@@ -24,13 +24,11 @@ public static class LighthouseHostFactory
...
-            // If overrides were provided to this method.
-            //if (!string.IsNullOrEmpty(ipAddress)) Environment.SetEnvironmentVariable("CLUSTER_IP", ipAddress);
-
-            //if (specifiedPort != null)
-            //    Environment.SetEnvironmentVariable("CLUSTER_PORT", specifiedPort.Value.ToString());
+            var argConfig = "";
+            if (ipAddress != null)
+                argConfig += $"akka.remote.dot-netty.tcp.public-hostname = {ipAddress}\n";
+            if (specifiedPort != null)
+                argConfig += $"akka.remote.dot-netty.tcp.port = {specifiedPort}";
```
Comment (eaba) [inline src/Lighthouse/LighthouseHostFactory.cs:31]: Is this reliable? What about this:
```csharp 
if (specifiedPort.HasValue)
                argConfig += $"akka.remote.dot-netty.tcp.port = {specifiedPort.Value}";
```

File: src/Lighthouse/LighthouseHostFactory.cs [hunk 2]
```diff
@@ -41,6 +39,11 @@ public static class LighthouseHostFactory
             if (useDocker)
                 clusterConfig = clusterConfig.BootstrapFromDocker();
 
+            // Values from method arguments should always win
+            if (!IsNullOrEmpty(argConfig))
```
Comment (eaba) [inline src/Lighthouse/LighthouseHostFactory.cs:43]: LGTM

**Review by eaba — COMMENTED** [review]
@Arkatufus I left you some comments

File: src/Lighthouse/LighthouseHostFactory.cs (hunk 1 above)
Comment (Arkatufus) [reply src/Lighthouse/LighthouseHostFactory.cs:31]: Its basically the same, its a nullable, it can be directly compared to a null

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**